src/main/scala/scalaz/nio/channels/AsynchronousChannel.scala

```diff
@@ -9,9 +9,9 @@ import java.nio.channels.{
   CompletionHandler => JCompletionHandler
 }
 
-import scalaz.nio.{ ByteBuffer, SocketAddress, SocketOption }
 import scalaz.nio.channels.AsynchronousChannel._
-import scalaz.zio.{ Async, ExitResult, IO }
+import scalaz.nio.{ ByteBuffer, SocketAddress, SocketOption }
+import scalaz.zio.{ Async, IO }
 import scalaz.{ IList, Maybe }
 
 import scala.concurrent.duration.Duration
```

Comment:
This should be changed to `scalaz.zio.Duration` and then calls to `timeout.length` and `timeout.unit` can be replaced by
`timeout.toNanos` and `java.util.concurrent.TimeUnit.NANOSECONDS`
or
`timeout.asJava.length` and `timeout.asJava.unit`